The ``add_integer`` module
==========================

Using ``add_integer``
-------------------------
->	add_integer(a, b=98)

Parameter 'b' is defaulted to 98
Parameters can be of int or float but floats are converted to integers

------------------------------

>>> add_integer = __import__('0-add_integer').add_integer

Basic Tests
    >>> add_integer(10)
    108
    >>> add_integer(-10)
    88
    >>> add_integer(10, 32)
    42
    >>> add_integer(20, -987)
    -967

Floats
    >>> add_integer(10.89)
    108
    >>> add_integer(12.34, 98.2)
    110

Type errors
    >>> add_integer("game")
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

    >>> add_integer(3+4j)
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

    >>> add_integer(None)
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

    >>> add_integer([3, 44])
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

    >>> add_integer(True)
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

    >>> add_integer({3+4j})
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

    >>> add_integer((3,))
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

    >>> add_integer(0, (23,))
    Traceback (most recent call last):
    ...
    TypeError: b must be an integer

    >>> add_integer(-10, "game")
    Traceback (most recent call last):
    ...
    TypeError: b must be an integer

    >>> add_integer(8, 3+4j)
    Traceback (most recent call last):
    ...
    TypeError: b must be an integer

    >>> add_integer(0, False)
    Traceback (most recent call last):
    ...
    TypeError: b must be an integer

    >>> add_integer(-10, {76})
    Traceback (most recent call last):
    ...
    TypeError: b must be an integer

    >>> add_integer(8, {2: 5})
    Traceback (most recent call last):
    ...
    TypeError: b must be an integer

    None:
        >>> add_integer()                       # doctest: +ELLIPSIS
        Traceback (most recent call last):
        ...
        TypeError: ...
